File: backend/app/schemas/marketplace.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class LocalityResponse(BaseModel):
    model_config = {"from_attributes": True}
    id: UUID
    name: str
    pin_code: str
    city: str
    state: str
    lat: str | None = None
    lng: str | None = None


class MarketplaceReportResult(BaseModel):
    result_type: str = "report"
    listing_id: str
    pin_code: str
    locality_name: str | None = None
    city: str
    property_type: str
    report_count: int
    latest_report_date: str | None = None
    vendor_name: str | None = None
    vendor_id: str | None = None
    vendor_tier: str | None = None
    avg_rating: float | None = None
    total_ratings: int = 0
    price: str | None = None
    latitude: str | None = None
    longitude: str | None = None


class MarketplaceVendorResult(BaseModel):
    result_type: str = "vendor"
    vendor_id: str
    vendor_name: str
    display_photo: str | None = None
    vendor_tier: str
    specialization_tags: list[str] | None = None
    avg_rating: float | None = None
    total_ratings: int = 0
    total_completed_jobs: int = 0
    avg_turnaround_hours: float | None = None
    quality_score: str = "0"
    service_areas: list[str] = []
    latitude: str | None = None
    longitude: str | None = None


class MarketplaceSearchResponse(BaseModel):
    results: list[MarketplaceReportResult | MarketplaceVendorResult]
    total: int
    page: int
    page_size: int

File: frontend/src/types/marketplace.ts
export interface MarketplaceReportResult {
  result_type: "report";
  listing_id: string;
  pin_code: string;
  locality_name: string | null;
  city: string;
  property_type: string;
  report_count: number;
  latest_report_date: string | null;
  vendor_name: string | null;
  vendor_id: string | null;
  vendor_tier: string | null;
  avg_rating: number | null;
  total_ratings: number;
  price: string | null;
  latitude: string | null;
  longitude: string | null;
}

export interface MarketplaceVendorResult {
  result_type: "vendor";
  vendor_id: string;
  vendor_name: string;
  display_photo: string | null;
  vendor_tier: string;
  specialization_tags: string[] | null;
  avg_rating: number | null;
  total_ratings: number;
  total_completed_jobs: number;
  avg_turnaround_hours: number | null;
  quality_score: string;
  service_areas: string[];
  latitude: string | null;
  longitude: string | null;
}

export type MarketplaceResult = MarketplaceReportResult | MarketplaceVendorResult;

export interface MarketplaceSearchResponse {
  results: MarketplaceResult[];
  total: number;
  page: number;
  page_size: number;
}

export interface LocalityOption {
  id: string;
  name: string;
  pin_code: string;
  city: string;
  state: string;
  lat: string | null;
  lng: string | null;
}
